packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1170,17 +1205,19 @@ export class MergeTree {
 					const locallyMovedSegments = this.locallyMovedSegments.get(localMovedSeq);
 
 					if (locallyMovedSegments) {
-						locallyMovedSegments.segments.forEach((segment: ISegmentLeaf) => {
-							segment.localMovedSeq = undefined;
+						for (const segment of locallyMovedSegments.segments) {
```

Comment:
The typing difference might be a good reason to revert this particular change and disable the lint rule (with a comment explaining why).